src/main/java/org/folio/service/loader/RecordLoaderServiceImpl.java

```diff
@@ -64,16 +66,35 @@ private void populateLoadResultFromSRS(List<String> uuids, JsonObject underlying
     loadResult.setInstanceIdsWithoutSrs(new ArrayList<>(singleInstanceIdentifiersSet));
   }
 
-  private List<JsonObject> populateLoadResultFromInventory(JsonObject instancesJson) {
-    List<JsonObject> instancesResult = new ArrayList<>();
-    JsonArray instances = instancesJson.getJsonArray("instances");
-    for (Object instance : instances) {
-      instancesResult.add(JsonObject.mapFrom(instance));
+  /**
+   * Method converts Json Response obtained from external services to a list of json Objects
+   * @param field field to read from the JsonObject
+   * @param jsonObject the response payload to convert
+   * @return
+   */
+  private List<JsonObject> populateLoadResultFromResponse(String field, JsonObject jsonObject) {
+    List<JsonObject> result = new ArrayList<>();
+    JsonArray jsonArray = jsonObject.getJsonArray(field);
+    LOGGER.debug("Populating result from external response {}", jsonArray);
+    for (Object instance : jsonArray) {
+      result.add(JsonObject.mapFrom(instance));
     }
-    return instancesResult;
+    return result;
   }
 
   private String getRecordContent(JsonObject record) {
     return record.getJsonObject("parsedRecord").getJsonObject("content").encode();
   }
+
+  @Override
+  public List<JsonObject> getHoldingsForInstance(String instanceId, OkapiConnectionParams params) {
+    Optional<JsonObject> optionalRecords = inventoryClient.getholdingsByInstanceId(instanceId, params);
+    return optionalRecords.map(holdings -> populateLoadResultFromResponse("holdingsRecords", holdings)).orElseGet(ArrayList::new);
+  }
+
+  @Override
+  public List<JsonObject> getAllItemsForHolding(List<String> holdingIds, OkapiConnectionParams params) {
+    Optional<JsonObject> optionalRecords = inventoryClient.getItemsByHoldingIds(holdingIds, params);
+    return optionalRecords.map(holdings -> populateLoadResultFromResponse("items", holdings)).orElseGet(ArrayList::new);
```

Comment:
I think "holdings" variable should be renamed to "items"

Reply:
Done